crates/rpc/rpc-engine-api/src/engine_api.rs

```diff
@@ -209,7 +211,13 @@ impl<Client: HeaderProvider + BlockProvider + StateProvider> EngineApi<Client> {
             return Ok(ForkchoiceUpdated::from_status(PayloadStatusEnum::Syncing))
         }
 
-        // TODO: update tip
+        if let Err(error) = self.forkchoice_state_tx.send(fork_choice_state) {
```

Comment:
is only the last received `fork_choice_state` valid and previously received `ForkchoiceState` are invalid?

because I believe `watch` only keeps the last received value.

Reply:
ye, that's correct. the state should be reset/promoted to whatever we get from the CL